pyrobosim/pyrobosim/core/world.py

```diff
@@ -624,19 +626,44 @@ def remove_all_objects(self, restart_numbering=True):
         if restart_numbering:
             self.object_instance_counts = {}
 
-    def update_bounds(self):
+    def update_bounds(self, entity, remove=False):
         """
         Updates the X and Y bounds of the world.
-
-        TODO: If we're just adding a single room, we only need to check that one
-        and there is probably a more efficient way to do this.
         """
-        for entity in itertools.chain(self.rooms, self.hallways):
+        if isinstance(entity, (Room, Hallway)):
             (xmin, ymin, xmax, ymax) = entity.polygon.bounds
-            self.x_bounds[0] = min(self.x_bounds[0], xmin)
-            self.x_bounds[1] = max(self.x_bounds[1], xmax)
-            self.y_bounds[0] = min(self.y_bounds[0], ymin)
-            self.y_bounds[1] = max(self.y_bounds[1], ymax)
+
+            if not self.x_bounds:
+                # When adding the first room
+                self.x_bounds = [xmin, xmax]
+                self.y_bounds = [ymin, ymax]
+                return
+
+            sets_x_bounds = (self.x_bounds[0] == xmin) or (self.x_bounds[1] == xmax)
+            sets_y_bounds = (self.y_bounds[0] == ymin) or (self.y_bounds[1] == ymin)
+            is_last_room = len(self.rooms) == 0 and isinstance(entity, Room)
```

Comment:
You should only need to do this check if `remove` is true. So just pop it in the if-statement below.